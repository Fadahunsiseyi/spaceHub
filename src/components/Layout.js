import { Outlet, Link } from "react-router-dom";

const Layout = () => {
  return (
        <>
        <container>
        <hearder> 
          <div><img src="./planet.png" width="40px"/></div>
          <h1>Space Traveler's Hub</h1>
        <nav>
        <ul>
          <li>
            <Link to="/">Rocket</Link>
          </li>
          <li>
            <Link to="/mission">Mission</Link>
          </li>
          <li>
            <Link to="/profile">MyProfile</Link>
          </li>
        </ul>
      </nav>
      </hearder>
      </container>
<hr></hr>
      <Outlet />
    </>
  )
};

export default Layout;