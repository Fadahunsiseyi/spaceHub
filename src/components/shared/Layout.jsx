import React from 'react';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';
import Header from './Header';

function Layout({ children }) {
  return (
    <div className="app-layout">
      <Helmet>
        <meta charSet="utf-8" />
        <title>Space Hub</title>
        <meta
          name="description"
          content="A web application for a companies that provide commercial and scientific space travel services"
        />
      </Helmet>
      <Header />
      <div className="content-container">{children}</div>
    </div>
  );
}

export default Layout;

Layout.propTypes = {
  children: PropTypes.element.isRequired,
};
