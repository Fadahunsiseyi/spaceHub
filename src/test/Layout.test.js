import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';
import Layout from '../components/shared/Layout';

describe('Pages renders correctly', () => {
  it('Should redenders NavBar Component succefully', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Layout />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
