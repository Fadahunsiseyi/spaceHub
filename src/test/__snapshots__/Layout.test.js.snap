// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Should redenders NavBar Component succefully 1`] = `
<div
  className="app-layout"
>
  <nav
    className="navbar"
  >
    <div
      className="brand-container"
    >
      <a
        aria-current="page"
        className="brand-wrapper active"
        href="#/"
        onClick={[Function]}
      >
        <img
          alt="Planet-img"
          className="brand"
          src="planet.png"
        />
        <h1
          className="nav-head"
        >
          Space Travelers' Hub
        </h1>
      </a>
    </div>
    <ul
      className="nav-menu"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="#/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="content-container"
  />
</div>
`;
